spritecontrol.cpp

```diff
@@ -105,8 +104,13 @@ bool SpriteChoreographer::should_change_pattern() {
 }
 
 void SpriteChoreographer::change_pattern() {
-	size_t rand = (size_t) (Noise::random() * m_enabled_patterns.size());
-	m_pattern = m_enabled_patterns.at(rand);
+	std::vector<PatternName> patterns = { };
+	std::copy_if(m_enabled_patterns.begin(), m_enabled_patterns.end(), std::back_inserter(patterns), [&](const PatternName &pattern) {
+		return pattern != m_pattern;
+	});
+
+	size_t rand = (size_t) (Noise::random() * patterns.size());
```

Comment:
Don't know why I didn't notice this at first but this probably shouldn't be called `rand` since it's undescriptive and it conflates with std::rand - `random_index` might be better.